server.js

```diff
@@ -1,30 +1,97 @@
 import express from "express";
 import cors from "cors";
+import expressListEndpoints from 'express-list-endpoints';
+import elves from "./data/elves.json";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
-const port = process.env.PORT || 8080;
+// Defines the port the app will run on. Defaults to 8080, but can be overridden when starting the server. Example command to overwrite PORT env variable value: PORT=1224 npm start
+const port = process.env.PORT || 1224; // Hoho
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+// Documentation endpoint
+app.get("/", (request, response) => {
+  const endpoints = expressListEndpoints(app);
+  response.json({
+    message: "Welcome to the Elves API! Here are the available endpoints:",
+    description: {
+      "/elves/all": "Get all elves",
+      "/elves/top-twelves": "Get the top twelves",
+      "/elves/titles/:title": "Get elves by title",
+      "/elves/:id": "Get a specific elf by ID",
+      "/test": "Test endpoint",
+    endpoints: endpoints
+    }
+  });
 });
 
-// Start the server
+/**
+ * Endpoint for getting all elves.
+ * This endpoint returns the complete list of elves from the elves.json.
+ */
+
+app.get("/elves/all", (request, response) => {
+  
+  // Return all elves
+  response.json(elves);
+});
```

Comment:
Hehe love the dataset! A more RESTful approach would be to name this endpoint just `/elves` and then have different query params to use as filters, e.g.
`/elves?top-twelves=true`
`/elves?title=backend dasher`